### PR TITLE
Fix bug where fine payment incorrectly shows Paid

### LINKED ISSUE #340: Fix bug where fine payment incorrectly shows 'Paid'
If there is no `paymentTypeDescription` the default return text value is 'Paid' even if the fee was `FORGIVEN`. We need to check the `paymentTypeID` as well.

### PR DESCRIPTION
**Pull request by jgreben**
Fixes #340 and #347

### CODE REVIEW

File: app/models/payment.rb [hunk 1]
```diff
@@ -42,6 +42,7 @@ def payment_date
 
   def resolution
     return fee_pay_info['paymentTypeDescription'] if fee_pay_info['paymentTypeDescription']
+    return fee_pay_info['paymentTypeID'] if fee_pay_info['paymentTypeID']
```
Comment (shelleydoljack) [inline app/models/payment.rb]: I'm assuming that we always have a paymentTypeID, so do we do anything with it (translate?)? And does the final condition ever get called (return Paid)?

Comment (jgreben) [reply app/models/payment.rb]: We refactored the payment model.

**Comment by jgreben** [comment]
Screen Shot 2019-07-30 at 1 07 53 PM

File: spec/models/payment_spec.rb [hunk 2]
```diff
@@ -58,12 +56,55 @@
...
+  context 'when payment resolution is a PAID_TYPE' do
     it 'has a resolution desctiption' do
-      expect(payment.resolution).to eq 'Payment using credit or debit card via MyAccount'
+      expect(payment.resolution).to eq 'Paid by credit card'
     end
+  end
 
-    it 'can tell if they paid their bill using a card' do
-      expect(payment).to be_paid_fee
+  context 'when no payment resolution is an UNPAID_TYPE' do
```
Comment (shelleydoljack) [inline spec/models/payment_spec.rb]: We should update the context to payment type.

**Pull request merged**